python/mlc_chat/compiler/compiler_pass/pipeline.py

```diff
@@ -5,13 +5,15 @@
 from tvm.relax import register_pipeline  # pylint: disable=no-name-in-module
 
 from ...support import logging
+from ..compile import CompileArgs
```

Comment:
I'm not sure if it's gonna be cyclic dependency

Reply:
Let's instead do this: in `compile.py`, we check if we really needs cublas dispatch using the following condition:
- `opt.cublas_gemm` is True
- `target.kind` is "cuda"
- `quantization` is `q0f16` or `q0f32`

If all of the above are True, pass in a flag `enable_cublas_gemm` which enables a pass called `CublasGemmDispatch(enabled=enable_cublas_gemm)`, which throws an error if `relax.ext.cublas` is not True.

Reply:
no it won't be cyclic dependency

Reply:
I want to hide the logic of whether to apply a pass inside the compiler pass file to keep other files simple. Other pass may also rely on `args`.

Reply:
The principle I'd love to keep here is to keep the compilation pipeline as independent as possible so that it could be used standalone in any projects. Imagine the usecase below:

```python
import mlc_chat.compiler

mod, ... = MyOwnLLM().export_tvm(...)
relax.build(mod, pipeline="mlc_llm", ...)
```

It can even be a single file that simply invokes our compilation pipeline, without knowing what exactly MLC LLM does.

Based on this principle, it would be the best to do our best to decouple. The proposal above, i.e. consolidate flag updates inside the dataclass `OptimizationFlags`, isn't that bad, compared with the existing implementation, because 1) `OptimizationFlags` does seem to be the place that collectively stores correct optimization flags, which is actually printed out on screen for users to better debug; 2) it doesn't really add lines of code or any implementation complexity; 3) and it helps with decoupling